### DISCUSSION TITLE
Any Authentication Alternatives

### DISCUSSION THREAD

**Discussion by eferay93** [tulir/whatsmeow#304] [General]
Dear Developers,

I am quite new to the repository and Golang. Thanks to your "mdtest", it was great to see the capabilities. I wonder if there is any alternative to authenticate to my whatsapp number? Currently, I go to my Whatsapp, and Settings>>Connected Devices>>Attach Device and I scan the QR code you generate.

**Comment by tulir** [comment]
No, that's the only way to log into WhatsApp web.

**Reply by eferay93** [reply]
Maybe the question would further be facilitated: Could we somehow emulate the traffic generated by whatsapp given that the QR we generate for its input to start authentication? Then, I guess it currently seems impossible and might be equivalent to break some sort of challange/response protocol or whatsoever used.

**Reply by Manjit2003** [reply]
Yes you can login with phone and OTP. Checkout the yowsup project.
